src/main/java/io/jenkins/plugins/audit/listeners/BuildStartListener.java

```diff
@@ -0,0 +1,56 @@
+package io.jenkins.plugins.audit.listeners;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.model.Cause;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.model.User;
+import hudson.model.listeners.RunListener;
+import org.apache.logging.log4j.audit.LogEventFactory;
+import io.jenkins.plugins.audit.event.BuildStart;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+
+@Extension
+public class BuildStartListener extends RunListener<Run> {
+    /**
+     * Fired when a build is started, event logged via Log4j-audit.
+     *
+     * @param run of type Run having the build information
+     * @param listener of type TaskListener that the onStarted method expects
+     */
+    @Override
+    public void onStarted(Run run, TaskListener listener) {
+        BuildStart buildStart = LogEventFactory.getEvent(BuildStart.class);
+
+        List causeObjects = run.getCauses();
+        List<String> causes = new ArrayList<>(causeObjects.size());
+        for (Object cause: causeObjects) {
+            Cause c = (Cause)cause;
+            causes.add(c.getShortDescription());
+        }
+
+        buildStart.setBuildNumber(run.getNumber());
+        buildStart.setCause(causes);
+        buildStart.setProjectName(run.getParent().getFullName());
+        buildStart.setTimestamp(new Date(run.getStartTimeInMillis()).toString());
+        User user = User.current();
+        if(user != null)
```

Comment:
@jvz Does it make sense to have the current user here? Won't this always be the `System`?

Reply:
If possible, we should try to extract the user from the list of Causes. If there are any `UserIdCause`s, those are useful for the userid. You might want to look at what other causes are available to see if you get any other ideas.